packages/paste-core/core-bundle/tools/generate.js

```diff
@@ -0,0 +1,62 @@
+const {join} = require('path');
+const {getRepoPackages} = require('../../../../tools/utils/getRepoPackages');
+const {writeToFile} = require('../../../../tools/utils/writeToFile');
+
+const BLACKLIST = ['@twilio-paste/core'];
+
+const CORE_BUNDLE_PATH = join(__dirname, '../');
+const CORE_BUNDLE_INDEX_PATH = join(CORE_BUNDLE_PATH, 'src/index.tsx');
+const CORE_BUNDLE_PACKAGE_PATH = join(CORE_BUNDLE_PATH, 'package.json');
+
+// Given a list of packages, output the index.tsx exports string
+function generateIndexFromPackageList(packageList) {
+  let output = '';
+  packageList.forEach(package => {
+    output = `${output}export * from '${package.name}';\n`;
+  });
+  return output;
+}
+
+// Given a list of packages, output the package.json dependencies field
+function generateDependenciesFromPackageList(packageList) {
+  const dependencies = {};
+  packageList.forEach(package => {
+    dependencies[package.name] = `^${package.version}`;
+  });
+  return dependencies;
+}
+
+// Given a list of packages, output only the packages that belong in 'core'
+function getAllCorePackages(packageList) {
+  return packageList.filter(item => {
+    const isCorePackage = item.location.includes('/paste-core/');
+    const isReleased = !item.private && (item.version !== '0.0.1' || item.version !== '0.0.0');
+    const isNotBlacklisted = !BLACKLIST.includes(item.name);
+
+    return isCorePackage && isReleased && isNotBlacklisted;
+  });
+}
+
+// Main
+(async () => {
```

Comment:
Reading this function explains the main procedure that happens right before a build is triggered on this package.

Reply:
Using private, version, and blacklisted seems like a good way to handle that filter. I like it.